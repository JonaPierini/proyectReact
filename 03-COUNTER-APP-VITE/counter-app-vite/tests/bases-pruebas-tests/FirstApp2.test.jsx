import { render, screen } from "@testing-library/react";
import FirsApp from "../../src/FirstApp";
describe("Prubas en FirsApp", () => {
  const titulo = "El tituuuuuulo";
  const subtitulo = "Subtiulo";
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<FirsApp titulo={titulo}></FirsApp>);
    //genera el snapshot
    expect(container).toMatchSnapshot();
  });
  test('debe de mostrar el mensaje "El tituuuuuulo"', () => {
    render(<FirsApp titulo={titulo}></FirsApp>);
    expect(screen.getByText(titulo)).toBeTruthy();
  });
  test("debe de mostrar el titulo en un h1", () => {
    render(<FirsApp titulo={titulo}></FirsApp>);
    //headin level 1 refiere a h1
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      titulo
    );
  });
  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<FirsApp titulo={titulo} subtitulo={subtitulo}></FirsApp>);
    //evalua la cantidad de subtitulos que tengos. getAllbyText es un arreglo
    expect(screen.getAllByText(subtitulo).length).toBe(1);
  });
});
