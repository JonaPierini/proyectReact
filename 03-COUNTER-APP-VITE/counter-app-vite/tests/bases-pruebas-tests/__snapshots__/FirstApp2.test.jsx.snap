// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas en FirsApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    El tituuuuuulo
  </h1>
  <h2>
     
     
  </h2>
  <p>
     
     
  </p>
</div>
`;
